### PR TITLE
*: force the new upgrading tidb to be owner 

### LINKED ISSUE #51269: the SQL in `sys.schema_unused_indexes` is over-simplified, and not aggregated across nodes
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
mysql> select * from sys.schema_unused_indexes;                                                                                                                                                                                    +---------------+-------------+------------+
| object_schema | object_name | index_name |
+---------------+-------------+------------+
| test          | t           | idx_1      |
| test          | t           | idx_2      |
| test          | t           | idx_1      |
| test          | t           | idx_2      |
+---------------+-------------+------------+
4 rows in set (0.00 sec)
```

An index in different nodes will appear multiple times in the `sys.schema_unused_indexes`. I forgot to finish the aggregation 🤦.

Re-write the SQL should work well:

```
CREATE OR REPLACE VIEW sys.schema_unused_indexes AS
		SELECT
			table_schema as object_schema,
			table_name as object_name,
			index_name
		FROM information_schema.cluster_tidb_index_usage
		WHERE
			table_schema not in ('sys', 'mysql', 'INFORMATION_SCHEMA', 'PERFORMANCE_SCHEMA') and
			index_name != 'PRIMARY'
		GROUP BY table_schema, table_name, index_name
		HAVING
			sum(last_access_time) is null;
```

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #51269

Problem Summary:
force the new upgrading tidb to be owner, so that owner can always use the new code during upgrade,
### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/session/bootstrap.go [hunk 1]
```diff
@@ -62,6 +63,8 @@ import (
 	"go.uber.org/zap"
 )
 
+var bootstrapOwnerKey = "/tidb/distributeLock/"
```
Comment (Benjamin2037) [inline pkg/session/bootstrap.go:66]:
```suggestion
var bootstrapOwnerKey = "/tidb/distributeDDLOwnerLock/"
```
Should this be more clear about the usage of this distribute lock.

File: pkg/session/bootstrap.go [hunk 2]
```diff
@@ -1369,6 +1372,77 @@ var (
...
+		logutil.BgLogger().Warn("etcd client is nil")
+		return
+	}
+	etcdSession, err := concurrency.NewSession(cli)
+	if err != nil {
+		return
+	}
+	mu := concurrency.NewMutex(etcdSession, bootstrapOwnerKey)
+	err = mu.Unlock(context.Background())
+	if err != nil {
```
Comment (Benjamin2037) [inline pkg/session/bootstrap.go:1419]: What if release lock meet error? Do we have method to make sure the lock always will be released successfully?

File: pkg/session/bootstrap.go [hunk 3]
```diff
@@ -1369,6 +1372,77 @@ var (
 	SupportUpgradeHTTPOpVer int64 = version174
 )
 
+func acquireLock(s sessiontypes.Session) bool {
```
Comment (tangenta) [inline pkg/session/bootstrap.go:1375]: The returning bool value is not used?

Comment (wjhuang2016) [reply pkg/session/bootstrap.go:1375]: I will check it

File: pkg/session/bootstrap.go (hunk 1 above)
Comment (Benjamin2037) [reply pkg/session/bootstrap.go:66]: Please also add comment for this variable‘s usage.

File: pkg/session/bootstrap.go (hunk 3 above)
Comment (Benjamin2037) [inline pkg/session/bootstrap.go:1375]: you have define returen of  acqurieLock, but caller did not handle the acquireLock fail case.

File: pkg/session/bootstrap.go [hunk 4]
```diff
@@ -1418,35 +1493,41 @@ func checkDistTask(s sessiontypes.Session, ver int64) {
...
+	var maxRetryCnt = 10
+	for ; i < maxRetryCnt; i++ {
+		ok := acquireLock(s)
+		if ok {
+			break
+		}
+	}
+	if i == maxRetryCnt {
+		logutil.BgLogger().Fatal("[upgrade] get ddl owner distributed lock failed", zap.Error(err))
+	}
```
Comment (Benjamin2037) [inline pkg/session/bootstrap.go:1513]: If acquire lock failed to max retry count，Here should we get version and do upgrade？

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by tangenta — APPROVED** [review]

**Review by yudongusa — APPROVED** [review]
LGTM

**Pull request merged**